docs/reference/UDS Core/IdAM/authentication-flows.md

```diff
@@ -0,0 +1,70 @@
+---
+title: Authentication Flow Customization
+tableOfContents:
+  maxHeadingLevel: 5
+---
+
+# Authentication Flow Customization
+
+:::note Environment variables configured in the [uds-core Keycloak values.yaml file](https://github.com/defenseunicorns/uds-core/blob/main/src/keycloak/chart/values.yaml#L30-32) have `REALM_` appended to them during creation. See [ Customization docs](https://uds.defenseunicorns.com/reference/uds-core/idam/customization/) for more information.:::
+
+:::warning If upgrading uds-core, be aware that Keycloak Admin manual configuration will probably be required to set new Realm values. See the clickops section below for how to do this. :::
+
+## Bundle Overrides
+To simplify the configuration of the available authentication flows, the following three environment variables have been exposed. These variables default to `true` in UDS Core, override their values in a bundle to disable.
+
+:::note These settings allow for enabling/disabling one or more of the Auth flows. Be aware that disabling all three will result in no options for registration and authentication (login). :::
+
+| Setting | Description | Options |
+| - | - | - |
+| [USERNAME_PASSWORD_AUTH_ENABLED](https://github.com/defenseunicorns/uds-core/blob/main/src/keycloak/chart/values.yaml#L30) | Toggle on/off the Username and Password Authentication flow. When disabled there will be no username password login, password / password confirm registration fields, no credential reset, and no update password options available. | `true`(default), `false` |
+| [X509_AUTH_ENABLED](https://github.com/defenseunicorns/uds-core/blob/main/src/keycloak/chart/values.yaml#L31) | Toggle on/off X509 (CAC) Authentication flow. | `true`(default), `false` |
+| [SOCIAL_AUTH_ENABLED](https://github.com/defenseunicorns/uds-core/blob/main/src/keycloak/chart/values.yaml#L32) | Toggle on/off Social (Google SSO, Azure AD, etc. ) Authentication flows.| `true`(default), `false` |
+
+These three variables handle the complexities of configuring the following environment variables, which are responsible for both visual (theme) and security (realm). The following variables are not exposed for overriding.
```

Comment:
I'm wondering if we should even have most of the doc after this (with the exception of the manual config)? If it's not exposed them I'm unsure if it's really providing value for the end user - but maybe it's more of a dev doc for our reference of all the toggles?

Reply:
per offline conversation, agreed the doc was not serving its original intent. Moved extra info to a new dev docs directory that will hopefully be a dumping ground of useful developer documentation and usecases that we won't maintain in the code.